lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -0,0 +1,75 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/http'
+
+module Datadog
+  module Contrib
+    # Rack module includes middlewares that are required to trace any framework
+    # and application built on top of Rack.
+    module Rack
+      # TraceMiddleware ensures that the Rack Request is properly traced
+      # from the beginning to the end. The middleware adds the request span
+      # in the Rack environment so that it can be retrieved by the underlying
+      # application. If request tags are not set by the app, they will be set using
+      # information available at the Rack level.
+      class TraceMiddleware
+        DEFAULT_CONFIG = {
+          tracer: Datadog.tracer,
+          default_service: 'rack'
```

Comment:
usually we're relying in the more settings to set other things like the `hostname` or the `port. Here I think it's better to stay clean. This is a low level instrumentation and the "traditional" way to use it, is together with another instrumentation (i.e. a library or a framework like Sinatra/Rails). If we provide a `hostname` here, it means that the tracer is configured twice and I don't like that, at some point, duplicated configuration can have weird side effects based on the initialization order (who configures the tracer first?).

As explained in the docs, if you want to use this middleware with custom configs, just use the `tracer.configure()` method before adding the middleware.